### DISCUSSION TITLE
Can't find documentation for  `quarkus-keycloak-admin-client` extension.

### DISCUSSION THREAD

**Discussion by omar-zahid** [quarkusio/quarkus#26836] [Q&A] [answered]
I tried looking for examples/documentation on keycloak admin client but there's nothing useful out there.
Should we even use this extension? I'm just trying to perform simple task such as listing all users in realm and can't do so.

Here's my example code.

```java
import org.keycloak.admin.client.Keycloak;
import org.keycloak.representations.idm.UserRepresentation;

@Path("/api/admin")
@Authenticated
@ApplicationScoped
public class AdminResource {

    @Inject
    Keycloak keycloak;

    @GET
    @Path("/users")
    @Produces(MediaType.APPLICATION_JSON)
    public List<UserRepresentation> getAllUsers() {
        return keycloak.realm("quarkus").users().list();

    }
}

```

Edit: Code above is not running. Getting build errors.

[a comment by an automated account is not shown]

**Comment by sberyozkin** [comment, marked as answer]
@omar-zahid This extension only exists to facilitate running Keycloak Admin Client in native mode, you just need to initialize  `Keycloak` the way you'd usually do according to the docs/examples in the Keycloak repository (I'm not sure I can see some obvious references though), here is how it is done in Quarkus test:
```
KeycloakBuilder.builder()
                .serverUrl(KEYCLOAK_SERVER_URL)
                .realm("master")
                .clientId("admin-cli")
                .username("admin")
                .password("admin")
                .build();
```

So the injection does not work. However, now that you've typed this code, I wonder if we can enhance this extension a bit and support this injection to simplify things...

But until then - use this extension if you'd like to use the admin client, just don't expect the injection working just yet

**Reply by omar-zahid** [reply]
Thanks. I'll give it a go. Appreciate it.

**Reply by sberyozkin** [reply]
@omar-zahid Np, I've opened #26837
